crates/fuel-core/src/state/rocks_db.rs

```diff
@@ -779,15 +779,20 @@ where
         let backup_dir = backup_dir.as_ref().join(Description::name());
         let backup_dir_path = backup_dir.as_path();
 
-        let backup_engine_options =
-            BackupEngineOptions::new(backup_dir_path).map_err(|e| {
+        let mut backup_engine_options = BackupEngineOptions::new(backup_dir_path)
+            .map_err(|e| {
                 DatabaseError::BackupEngineInitError(anyhow::anyhow!(
                     "Couldn't create backup engine options for path `{}`: {}",
                     backup_dir_path.display(),
                     e
                 ))
             })?;
 
+        let cpu_number =
+            i32::try_from(num_cpus::get()).expect("The number of CPU can't exceed `i32`");
```

Comment:
In 30 years someone will run this code on a typical gigacore laptop and become very angry at this assumption :joy: